be/src/exec/olap_common.h

```diff
@@ -312,13 +317,13 @@ class ColumnValueRange {
         _high_value = TYPE_MAX;
         _low_op = FILTER_LARGER_OR_EQUAL;
         _high_op = FILTER_LESS_OR_EQUAL;
-        _contain_null = true;
+        _contain_null = _is_nullable_col;
     }
 
     bool is_whole_value_range() const {
         return _fixed_values.empty() && _low_value == TYPE_MIN && _high_value == TYPE_MAX &&
                _low_op == FILTER_LARGER_OR_EQUAL && _high_op == FILTER_LESS_OR_EQUAL &&
-               contain_null();
+               _is_nullable_col == contain_null();
```

Comment:
which case `_is_nullable_col` = false and  `contain_null()` = true

Reply:
I don't think this will happen, here add a DCHECK